idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/ui/toolwindow/ToolWindowSyntax.kt

```diff
@@ -0,0 +1,310 @@
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.MessageType
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ToolWindowAnchor
+import com.intellij.openapi.wm.ToolWindowManager
+import java.awt.BorderLayout
+import java.awt.LayoutManager
+import javax.swing.Icon
+import javax.swing.JComponent
+import javax.swing.JPanel
+import javax.swing.event.HyperlinkEvent
+
+/**
+ * Tool windows have several use-cases, though there used in two different scenario. Either, to display content resulting from an computation, which [ToolWindowSyntax] materializes with
+ * [addToolWindowWithAction] and [registerToolWindow], or establishing a 'persistently' visible tool window the user can interact with at all time. Please refer to this
+ * [link](http://www.jetbrains.org/intellij/sdk/docs/user_interface_components/tool_windows.html?search=tool) for the second scenario.
+ */
+interface ToolWindowSyntax {
+
+  /**
+   * registers an tool window with displayName [toolId] and activates it by invoking the Action with the title [toolId].
+   * The following example enables the internal `Fir Explorer`, whenever the current document is a kotlin file.
+   * ```kotlin:ank:playground
+   * import arrow.meta.Plugin
+   * import arrow.meta.ide.IdeMetaPlugin
+   * import arrow.meta.ide.resources.ArrowIcons
+   * import arrow.meta.invoke
+   * import com.intellij.openapi.actionSystem.CommonDataKeys
+   * import org.jetbrains.kotlin.idea.KotlinFileType
+   * import org.jetbrains.kotlin.idea.actions.internal.FirExplorerToolWindow
+   *
+   * val IdeMetaPlugin.exampleToolWindow: Plugin
+   *   get() = "ShowFirInTheIde" {
+   *     meta(
+   *       addToolWindowWithAction(
+   *         toolId = "Show Fir",
+   *         actionId = "Unique",
+   *         icon = ArrowIcons.ICON4,
+   *         content = { project, toolWindow ->
+   *           FirExplorerToolWindow(project, toolWindow)
+   *         },
+   *         update = { e ->
+   *           e.presentation.isEnabled = e.project != null && e.getData(CommonDataKeys.PSI_FILE)?.fileType == KotlinFileType.INSTANCE
+   *         }
+   *       )
+   *     )
+   *   }
+   * ```
+   * @param content these extensions allow you to define JComponents [toolWindowWithEditor], [toolWindowContent], [simpleWorkSpace], [toolWindowWithEditor] or create your own costume implementation.
+   * @param anchor where the tool window is located at ide start
+   * @param actionId needs to be unique
+   */
+  fun IdeMetaPlugin.addToolWindowWithAction(
+    toolId: String,
+    actionId: String,
+    icon: Icon,
+    content: (Project, ToolWindow) -> JComponent,
+    canCloseContent: Boolean = false,
+    anchor: ToolWindowAnchor = ToolWindowAnchor.RIGHT,
+    isLockable: Boolean = false,
+    update: (AnActionEvent) -> Unit = Noop.effect1
+  ): ExtensionPhase =
+    addAnAction(
+      actionId,
+      anAction(
+        toolId,
```

Comment:
the `toolId` is both the title and technical id. The user can index the tool window with the `toolId`, which is what seems more intuitive. Simplifying it to `id` rather than `title` made unregistering a tool window less complicated, regarding to identify the technical id.